src/account_manager/operations/account_recovery.rs

```diff
@@ -89,11 +103,13 @@ impl AccountManager {
             let mut tasks = Vec::new();
             for _ in 0..account_gap_limit {
                 let mut new_account = self.create_account();
+                let sync_options_ = sync_options.clone();
                 tasks.push(async move {
                     tokio::spawn(async move {
                         let new_account = new_account.finish().await?;
-                        let account_outputs_count =
-                            new_account.search_addresses_with_outputs(address_gap_limit).await?;
+                        let account_outputs_count = new_account
+                            .search_addresses_with_outputs(address_gap_limit, sync_options_.clone())
```

Comment:
Is this clone necessary ?

Reply:
I think so

Reply:
Are you sure ? I removed it and it's fine

Reply:
Ah you're right I thought you meant the clone to create sync_options_